pkg/osv/osv.go

```diff
@@ -306,29 +291,54 @@ func HydrateWithClient(resp *BatchedResponse, client *http.Client) (*HydratedBat
 	return &hydrated, nil
 }
 
-// makeRetryRequest will return an error on both network errors, and if the response is not 200
+// makeRetryRequest executes HTTP requests with exponential backoff retry logic
 func makeRetryRequest(action func() (*http.Response, error)) (*http.Response, error) {
-	var resp *http.Response
-	var err error
+	const maxRetryAttempts = 4
+	const jitterMultiplier = 2
+
+	var lastErr error
 
 	for i := range maxRetryAttempts {
-		// rand is initialized with a random number (since go1.20), and is also safe to use concurrently
-		// we do not need to use a cryptographically secure random jitter, this is just to spread out the retry requests
-		// #nosec G404
-		jitterAmount := (rand.Float64() * float64(jitterMultiplier) * float64(i))
-		time.Sleep(time.Duration(i*i)*time.Second + time.Duration(jitterAmount*1000)*time.Millisecond)
-
-		resp, err = action()
-		if err == nil {
-			// Check the response for HTTP errors
-			err = checkResponseError(resp)
-			if err == nil {
+		resp, err := action()
+		if err != nil {
+			lastErr = fmt.Errorf("attempt %d: request failed: %w", i+1, err)
+			if i == maxRetryAttempts-1 {
 				break
 			}
+			backoff := time.Duration(i*i) * time.Second
```

Comment:
It's not clear why jitter is changed here, the new implementation requires duplicating the jitter code? Can you revert this to the previous implementation